app/views/play/level/ControlBarView.coffee

```diff
@@ -38,13 +43,37 @@ module.exports = class ControlBarView extends CocoView
     @levelID = @levelSlug or @level.id
     @spectateGame = options.spectateGame ? false
     @observing = options.session.get('creator') isnt me.id
+
+    @levelNumber = ''
+    if @level.get('type') is 'course' and @level.get('campaignIndex')?
+      @levelNumber = @level.get('campaignIndex') + 1
+    if @courseInstanceID
+      @courseInstance = new CourseInstance(_id: @courseInstanceID)
+      @supermodel.trackRequest(@courseInstance.fetch().then(=>
+        @classroom = new Classroom(_id: @courseInstance.get('classroomID'))
+        @supermodel.trackRequest @classroom.fetch()
+      ))
```

Comment:
Until we upgrade to jQuery 3, we should avoid using jQuery promises, since they're [confusingly different than standard promises](https://github.com/kriskowal/q/wiki/Coming-from-jQuery). To do that, wrap an uncalled jQuery `then` with a `new Promise`.

```
request = new Promise(@courseInstance.fetch().then)
@supermodel.trackRequest(request.then(=>
  @classroom = new Classroom(_id: @courseInstance.get('classroomID'))
  @supermodel.trackRequest @classroom.fetch()
))
```